borg/constants.py

```diff
@@ -12,8 +12,14 @@
 CACHE_TAG_NAME = 'CACHEDIR.TAG'
 CACHE_TAG_CONTENTS = b'Signature: 8a477f597d28d172789f06886806bc55'
 
-DEFAULT_MAX_SEGMENT_SIZE = 5 * 1024 * 1024
-DEFAULT_SEGMENTS_PER_DIR = 10000
+# A large, but not unreasonably large segment size. Always less than 2 GiB (for legacy file systems). We choose
+# 500 MiB which means that no indirection from the inode is needed for typical Linux file systems.
+# Note that this is a soft-limit and can be exceeded (worst case) by a full maximum chunk size and some metadata
+# bytes. That's why it's 500 MiB instead of 512 MiB.
+DEFAULT_MAX_SEGMENT_SIZE = 500 * 1024 * 1024
```

Comment:
In case we'll ever have to do a segment-converter for upgrading, we'ld have to deal with these huge segment files (and could run out of memory when trying to read them into memory for processing) - that would not be a problem if we just increase to 50 or 100MB.

So the question is whether the speedup of the  50/100 -> 500 MB segment file increase rectifies that potential trouble. I'ld guess the speedup from 5 -> 50 is significant, while the speedup from 50 -> 500 might be negligible.